doc/source/rllib/connector-v2.rst

```diff
@@ -0,0 +1,596 @@
+                and len(input_observation_space.shape) == 1
+            )
+
+            # This connector concatenates the last 10 observations at axis=0, so the
+            # output space has a shape of (10*N,).
+            return gym.spaces.Box(
+                low=input_observation_space.low,
+                high=input_observation_space.high,
+                shape=(input_observation_space.shape[0] * 10,),
+                dtype=input_observation_space.dtype,
+            )
+
+        def __call__(self, *, rl_module, batch, episodes, **kwargs):
+            # Assume that the input `batch` is empty. Note that this may not be the case
+            # if you have other custom connector pieces before this one.
+            assert not batch
+
+            # Loop through all (single-agent) episodes.
+            for single_agent_episode in self.single_agent_episode_iterator(episodes):
+                # Get the 10 most recent observations from the episodes.
+                last_10_obs = single_agent_episode.get_observations(
+                    indices=[-10, -9, -8, -7, -6, -5, -4, -3, -2, -1], fill=0.0
+                )
+                # Concatenate the two observations.
+                new_obs = np.concatenate(last_10_obs, axis=0)
+
+                # Add the new observation to the `batch` using the
+                # `ConnectorV2.add_batch_item()` utility.
+                self.add_batch_item(
+                    batch=batch,
+                    column=Columns.OBS,
+                    item_to_add=new_obs,
+                    single_agent_episode=single_agent_episode,
+                )
+
+                # Note that we do not write the stacked observations back into the episode
+                # as this would interfere with the next call of this same connector (it
+                # would try to stack already stacked observations and thus produce a shape error).
+
+            # Return batch (with stacked observations).
+            return batch
+
+
+Now since the returned `batch` in the above env-to-module piece is discarded after the model forward pass
+(and not stored in the episodes), we have to make sure to perform the framestacking again on the Learner
+side of things.
+
+
+.. tip::
+    There are already off-the-shelf ConnectorV2 pieces available to users. These perform the task of
+    stacking the last `N` observations in both the env-to-module and Learner pipelines:
+
+    .. code-block:: python
+
+        from ray.rllib.connectors.common.frame_stacking import FrameStacking
+
+        # Framestacking on the EnvRunner side.
+        config.env_runners(
+            env_to_module_connector=lambda env: FrameStacking(num_frames=N),
+        )
+        # Then again on the Learner side.
+        config.training(
+            learner_connector=lambda obs_space, act_space: FrameStacking(num_frames=N, as_learner_conector=True),
+        )
+
+
+
+
+Point to more examples in the `examples/connectors` folder.
```

Comment:
This is a note to self it appears.

If we already have advanced examples in a folder, these examples should also be there, yes?